src/components/BarChart.js

```diff
@@ -0,0 +1,56 @@
+import { extent, max } from 'd3-array'
+import { scaleLinear, scaleTime } from 'd3-scale'
+import { timeParse } from 'd3-time-format'
+import React from 'react'
+
+import XAxis from './XAxis'
+import YAxis from './YAxis'
+
+const BarChart = ({
+  size = { width: 600, height: 400 },
+  margin = { top: 20, right: 20, bottom: 30, left: 50 },
+  labels = true,
+  data,
+}) => {
+  const height = size.height - margin.top - margin.bottom
+  const parse = timeParse('%Y-%m-%d')
+  const dataClean = data.map(d => ({ date: parse(d[0]), value: +d[1] }))
+  const width = size.width - margin.left - margin.right
+
+  const x = scaleTime()
```

Comment:
ok, so after changing to scaleBand, it looks like a couple tweaks to XAxis need to be made:

```
  const format = scale.tickFormat ? scale.tickFormat.apply(scale) : x => String(x)
  const values = scale.ticks ? scale.ticks(tickCt) : scale.domain()
```

finally, i think u can make the rect width more dynamic (`width={x.bandwidth()}`)

should look something like this (the labels look off but that's really just because they're so freaking long):
![image](https://cloud.githubusercontent.com/assets/1060893/21038783/9457a19e-bda6-11e6-9c34-f71c936c8346.png)